### DISCUSSION TITLE
Evaluating covariance of free energy (df_i) from f_i

### DISCUSSION THREAD

**Discussion by karanv99** [choderalab/pymbar#487] [General]
Hello everyone, I am using the discussion forum for the first time. So if I am doing a mistake kindly excuse me. I am using pymbar to evaluate pmf for my REMD simulation. 

My doubt was in using the command mbar.computePMF(). The max value of covariance evaluated with the computePMF command df_i is 1. 

0 <= df_i <= 1

But covariance does not have any limits on its value. Could someone please explain why it is so? I might be very wrong and doing a horrible mistake, kindly excuse me if I am doing so. 

Thank you

**Comment by mrshirts** [comment]
Hi, @karanv99 . It's a little hard to see if it makes sense or not without the output in many cases, the uncertainty may indeed have a max at 1. 

Note that computePMF has been replaced with computeFES in pymbar4.0, which has better behavior (that is, it is better controllable and better documented).

**Comment by karanv99** [comment]
Thank you for replying @mrshirts, I have listed about the problem faced in detail. It would be really helpful to hear from you. Thank you.


**PROBLEM 1**

Evaluated the PMF with the following command:

results = mbar.computePMF(u\_kn, bin\_kn, nbins, uncertainties='from-lowest', return\_dict=True)

This gave me **f\_i** and **df\_i** which are free energies and covariance in free energies at each of the 25 bins (shown below).

<img width="416" alt="Screenshot 2022-12-28 at 9 17 25 PM" src="https://user-images.githubusercontent.com/101196030/210575780-f81056a8-26bc-43e1-94cc-255c268ca365.png">


The free energies evaluated in bins 14, 15, 16, 22, 23, 24 are very huge due to less sampling. But the covariance evaluated of these bins are capped at 1. Shouldn't covariance be greater than 1 in these bins? Because the free energy values do not make sense?


**PROBLEM 2**


Evaluated the PMF with order parameter(alpha-helicity) and got the following free energies. Bin 0 has only 342 configurations, which suggests that particular configuration is not stable and hence not sampled. But the free energy is coming out as the least (due to uncertainties='from-lowest')

<img width="414" alt="Screenshot 2022-12-28 at 9 48 14 PM" src="https://user-images.githubusercontent.com/101196030/210576058-20fca85a-62af-4fd4-97af-da175413c821.png">


**PROBLEM 3**

Pymbar 4.0 uses jax.numpy and jax is not well configured for newer macs(m1) and I tried using it on an intel i7 cpu but it is not getting initialized. Pymbar 3.0 versions are consistent and works for me.

**Comment by mrshirts** [comment]
Pymbar 3.0 is not generally going to be supported in the future - I will try to identify the issues, but can't guarantee 3.0 will fix any issues found.

Not being able to install pymbar 4.0 is more worrisome, as we want to enable that.  Can you reply to #488 and add more information as to what happens when installing pymbar 4.0 on M1 macs, and on intel I7?

**Comment by mrshirts** [comment]
To identify any bug with the df=1.00, I will need a sample data set to work with - It's too hard to inspect what is going on without data. 

In terms of problem 2, "Bin 0 has only 342 configurations, which suggests that particular configuration is not stable and hence not sampled."  Is the simulation starting in this configuration?  Did you remove out of equilibrium samples at the beginning of the simulation?  If out-of-equilibrium samples are included, there is a high likelihood that the results will be incorrect.
